### ISSUE TITLE
require-atomic-updates false positive

### ISSUE THREAD

**Issue by TheModMaker** [eslint/eslint#11687]
May be related to #11194, but that looked more like naming conflicts.

**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 11.13.0
* **npm Version:** 6.7.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "parserOptions": {
        "ecmaVersion": 2017
    },
    "rules": {
        "require-atomic-updates": "error",
    },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

https://eslint.org/demo/#eyJ0ZXh0IjoiLyplc2xpbnQgcmVxdWlyZS1hdG9taWMtdXBkYXRlczogZXJyb3IgKi9cclxuXHJcbmFzeW5jIGZ1bmN0aW9uIGYoKSB7XHJcbiAgICB0cnkge1xyXG4gICAgICAgIHRoaXMuZm9vID0gZG9Tb21ldGhpbmcoKTtcclxuXHR9ICBjYXRjaCAoZSkge1xyXG4gICAgICAgIHRoaXMuZm9vID0gbnVsbDtcclxuICAgICAgICBhd2FpdCBkb0Vsc2UoKTtcclxuICAgIH1cclxufVxyXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjgsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=

```js
async function f() {
  try {
    this.foo = doSomething();
  }  catch (e) {
    this.foo = null;
    await doElse();
  }
}
```

**What did you expect to happen?**
No errors.

**What actually happened? Please include the actual, raw output from ESLint.**

```
foo.js
  5:5  error  Possible race condition: `this.foo` might be reassigned based on an outdated value of `this.foo`  require-atomic-updates

✖ 1 problem (1 error, 0 warnings)
```

The `doSomething` method is synchronous so it cannot violate this rule about using "await".  I'm thinking it is confused about reassigning in the "catch" block, but for a synchronous method, that should be allowed and not violate this rule.

**Are you willing to submit a pull request to fix this bug?**
No.

**Comment by mysticatea** [comment]
Thank you for the report.

I confirmed that #11224 fixes this issue also in fortune.
